### ISSUE TITLE
Robot Framework is incompatible with OpenJDK 17 and 18 

### ISSUE THREAD

**Issue by csantune** [robotframework/robotframework#4100]
Hi,

I test several versions of robotframework-<version>.jar but it always fails with this error

```
$ java -jar robotframework-4.1.jar 
Exception in thread "MainThread" Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/data/poc/robotframework-4.1.jar/Lib/robot/__init__$py.class", line 44, in <module>
  File "/data/poc/robotframework-4.1.jar/Lib/robot/rebot$py.class", line 45, in <module>
  File "/data/poc/robotframework-4.1.jar/Lib/robot/run$py.class", line 44, in <module>
  File "/data/poc/robotframework-4.1.jar/Lib/robot/running/__init__$py.class", line 104, in <module>
  File "/data/poc/robotframework-4.1.jar/Lib/robot/running/userkeyword$py.class", line 24, in <module>
  File "/data/poc/robotframework-4.1.jar/Lib/robot/running/userkeywordrunner$py.class", line 29, in <module>
  File "/data/poc/robotframework-4.1.jar/Lib/robot/running/timeouts/__init__$py.class", line 29, in <module>
  File "/data/poc/robotframework-4.1.jar/Lib/robot/running/timeouts/posix$py.class", line 16, in <module>
ImportError: cannot import name setitimer

$ java --version
openjdk 17-ea 2021-09-14
OpenJDK Runtime Environment (build 17-ea+19-Ubuntu-1ubuntu1)
OpenJDK 64-Bit Server VM (build 17-ea+19-Ubuntu-1ubuntu1, mixed mode, sharing)
```

I try to build keyword in java,  jython and robotframework.jar are they mandatory ?

**Comment by csantune** [comment]
Related to the old issue https://github.com/robotframework/robotframework/issues/3050

**Comment by csantune** [comment]
Related to the old issue https://github.com/robotframework/robotframework/issues/3050
It works with the JRE 8

**Comment by pekkaklarck** [comment]
The problem occurs when importing `robot.running.timeouts.posix` module which shouldn't be imported at all when using Jython. This got me thinking the problem could be in our Jython detection but everything worked fine with OpenJDK 11 that I had installed.

I was, however, able to reproduce the problem with OpenJDK 17 and sure enough the issue is in Jython detection. We expect `sys.platform` to be in format `java<major>.<minor>.<patch><whatever>` when using Jython, but with OpenJDK 17 it's only `java17`. Fixing this ought to be easy.

Luckily this was reported just a day before RF 4.1.2, the last planned release to support Python 2 and Jython, and we can easily get this included. As part of debugging this I also encountered the same problem as in #3780 and hopefully get that fixed as well.

**Comment by pekkaklarck** [comment]
The problem isn't limited to the standalone JAR, same problem occurs also when using normal Jython on OpenJDK 17 and 18. 

OpenJDK 16 that I got from https://jdk.java.net/16 still works fine. Could be that this is just due to its version actually being 16.0.2 while OpenJDK 17 and 18 have no minor or patch level version numbers.

**Comment by danielleberre** [comment]
I experienced the issue with Java 11 and Java 15 on macOS, so I guess that the issue is not limited to Java 17 and 18.

**Comment by pekkaklarck** [comment]
As a strange coincidence, #4099 that explains this exact same problem was reported just few hours before this issue. That and the old #3050 demonstrate there were several people affected by this. Somewhat unfortunate that #3050 was originally closed by the original submitted as a configuration issue and I didn't pay enough attention to it either. Anyway, great that the bug is now fixed and fix can be included in RF 4.1.2, the last planned Jython compatible release.

**Comment by csantune** [comment]
We don't work together just a strange coincidence :)
